build/Dockerfile

```
@@ -0,0 +1,101 @@
+# (c) Cartesi and individual authors (see AUTHORS)
+# SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+# syntax=docker.io/docker/dockerfile:1.4
+
+ARG RUST_VERSION=1.71.0
+ARG SERVER_MANAGER_VERSION=0.8.2
+ARG ROLLUPS_CONTRACTS_VERSION=1.0.2
+
+ARG BASE_PATH=/opt/cartesi
+ARG RUST_BUILD_PATH=${BASE_PATH}/src/rollups-node/offchain
+ARG DEPLOYMENT_PATH=${BASE_PATH}/share/deployments
+ARG RUNTIME_DIR=/var/opt/cartesi
+
+#
+# On-chain deployment files
+#
+FROM debian:bookworm-slim as deployment
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y curl
+
+ARG DEPLOYMENT_PATH
+WORKDIR ${DEPLOYMENT_PATH}
+
+# Download deployment files from npm
+ARG ROLLUPS_CONTRACTS_VERSION
+RUN curl "https://registry.npmjs.org/@cartesi/rollups/-/rollups-${ROLLUPS_CONTRACTS_VERSION}.tgz" \
+        -o /tmp/rollups.tgz
+RUN tar zxf /tmp/rollups.tgz -C /tmp
+RUN cp /tmp/package/export/abi/* .
+
+#
+# Cargo chef
+#
+FROM rust:${RUST_VERSION}-bookworm AS rust-chef
+
+RUN rustup component add rustfmt
+RUN cargo install cargo-chef
+
+# The workdir will be inherited by the following Rust images
+ARG RUST_BUILD_PATH
+WORKDIR ${RUST_BUILD_PATH}
+
+#
+# Cargo chef prepare stage
+#
+FROM rust-chef as rust-planner
+
+COPY ./offchain/ .
+RUN cargo chef prepare --recipe-path recipe.json
+
+#
+# Cargo chef cook stage
+#
+FROM rust-chef as rust-builder
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        cmake libprotobuf-dev protobuf-compiler curl
+
+# Build dependencies with cargo chef
+COPY --from=rust-planner ${RUST_BUILD_PATH}/recipe.json .
+RUN cargo chef cook --release --recipe-path recipe.json
+
+# Build application
+COPY ./offchain/ .
+RUN cargo build --release
+
+#
+# Runtime
+#
+FROM cartesi/server-manager:${SERVER_MANAGER_VERSION} as rollups-node
+
+USER root
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libpq5 ca-certificates curl
```

I don't see `curl` being used during build, so we could remove. Unless you have plans to use it at runtime.

```suggestion
    libpq5 ca-certificates
```

---

It is used internally by the Rust build to download the contract JSON.